cmd/devenv/tunnel/tunnel.go

```diff
@@ -86,12 +87,30 @@ func (o *Options) Run(ctx context.Context) error { //nolint:funlen
 		return err2
 	}
 
+	var elevatedPermissions bool
+	go func(ctx context.Context) {
+		// Sleep for 7 seconds the first time before checking to alert for permissions.
+		async.Sleep(ctx, time.Second*7)
+		for {
+			if elevatedPermissions {
+				// Got elevated permissions, don't alert.
+				return
+			}
+
+			alert.Alert("Your tunnel needs elevated permissions!")
+
+			// Sleep for 15 seconds between each alert after the first one.
+			async.Sleep(ctx, time.Second*15)
+		}
+	}(ctx)
+
```

Comment:
This is not thread safe.  I doubt it matters for this specific tool but the thread safe version is like so (also added a ctx.Err() == nil check):


```suggestion
	var elevatedPermissions int32
	go func(ctx context.Context) {
		// Sleep for 7 seconds the first time before checking to alert for permissions.
		async.Sleep(ctx, time.Second*7)
		for atomic.LoadInt32(&elevatedPermissions) == 0 && ctx.Err() == nil {
			alert.Alert("Your tunnel needs elevated permissions!")

			// Sleep for 15 seconds between each alert after the first one.
			async.Sleep(ctx, time.Second*15)
		}
	}(ctx)

```

Reply:
You'd have to replace the `elevatedPermissions = true` with `atomic.AddInt32(&elevatedPermissions, 1)`

Reply:
Actually, this is a clear use of signaling, so an alternate approach without using atomic would be via using a channel for signaling:

```golang
var done chan struct{}
go func(ctx context.Context) {
    async.Sleep(time.Second*15)
    for ctx.Err() != nil {
        alert.Alert("...")
        select {
        case <- ctx.Done():  return
        case <- time.After(time.Seconds*15):
        }
    }
}(ctx)

...
close(done)
```

Reply:
done